content/docs/user-documentation/configuring.md

```diff
@@ -10,7 +10,69 @@ summary: Configuration options with MicroShift
 modified: "2021-11-03T16:24:11.538+01:00"
 ---
 
-## WIP Content coming soon
+<!-- 
+todo: 
+- logging 
+-->
 
-- config.yaml vs env vars vs command line
-- logging
+## Configuration
+
+Microshift can be configured in three simple ways, in order of precedence: 
+- Commandline arguments
+- Environment variables
+- Configuration file
+
+Values are passed in to modify the `MicroshiftConfig` defined in `pkg/config/config.go`. 
+
+#### Configuration File
```

Comment:
Structuring by configuration method may become unwieldy and repetitive when we introduce more params.

How about instead listing config params in a table with columns "Command Line Arg", "Env. Variable", "Config File", "Meaning", "Default"?
Then, we could introduce structure by introducing sections per logical grouping of config params, e.g. "Logging", each section having a table like that.

Reply:
That's a really good idea. Would there be anyway that we could tie that in with the implementation to have stronger cohesion as more changes are made?

Reply:
@fzdarsky I feel like there are two problems with the approach of maintaining a list like that:
1. The Config options are in a separate repo, whereas the docs are in an entirely separate repo. This creates extra effort needed to quickly make changes and have them update.
2. Not all of the params are necessarily available to be changed via environment variable or CLI option, so maintaining this will be difficult if this gets changed.

Reply:
> 1. The Config options are in a separate repo, whereas the docs are in an entirely separate repo. This creates extra effort needed to quickly make changes and have them update.

Yes, but this applies equally whether the doc is structured by config method or group of config params, or?
Also, both MicroShift engs and the team eventually owning docs may want to make a deliberate decision on which configs to add.

> 2. Not all of the params are necessarily available to be changed via environment variable or CLI option, so maintaining this will be difficult if this gets changed.

Params should be configurable via all 3 methods, so probably a gap we should close.

Reply:
Okay. In that case, should we merge this PR or wait until we can consolidate it with the rest of the docs?